index.js

```diff
@@ -144,22 +145,49 @@ module.exports = function (babel) {
         return newPath.node ? [newPath.get('callee')] : [];
       });
     }
+
+    if (state.opts.ensureModuleApiPolyfill) {
+      processModuleApiPolyfill(state);
+    }
+  }
+
+  let allAddedImports = Object.create(null);
+
+  function processModuleApiPolyfill(state) {
+    for (let module in allAddedImports) {
+      let addedImports = allAddedImports[module];
+
+      for (let addedImport in addedImports) {
+        let { path } = addedImports[addedImport];
+
+        if (path && path.node) {
+          processImportDeclaration(t, path, state);
+
+          if (path.removed) {
+            addedImports[addedImport] = undefined;
+          }
+        }
+      }
+    }
   }
 
   let visitor = {
     Program(path, state) {
+      if (state.opts.ensureModuleApiPolyfill) {
+        // Setup state for the module API polyfill
+        setupState(t, path, state);
```

Comment:
Have you confirmed that our `ensureImport` methods are compatible? Specifically, I'm worried about 

https://github.com/ember-cli/babel-plugin-ember-modules-api-polyfill/blob/fffc8dd3123a275861c9b6e85dc2db041b39acf2/src/index.js#L90

It **always** clobbers `state.ensureImport`, which (I think) means that our ensureImport will be gone and if we have any custom logic there it will be lost (including the `addedImports` being updated properly).

Reply:
Our `ensureImport` is added later, and so ours clobbers theirs actually. They are compatible, as the tests show, though I do agree this is a bit worrisome. I don't think it should be an issue in the short term, but if we want to be extra safe, we could try to make the local name more unique.